benchmark/torch/ddpg/train.py

```diff
@@ -0,0 +1,173 @@
+#   Copyright (c) 2020 PaddlePaddle Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import numpy as np
+import random
+import torch
+import gym
+import argparse
+from parl.utils import logger, tensorboard, ReplayMemory
+from mujoco_model import MujocoModel
+from mujoco_agent import MujocoAgent
+from parl.algorithms import DDPG
+
+
+# Run episode for training
+def run_train_episode(agent, env, rpm, args):
+    action_dim = env.action_space.shape[0]
+    max_action = float(env.action_space.high[0])
+    obs, done = env.reset(), False
+    episode_reward, episode_steps = 0, 0
+    while not done:
+
+        episode_steps += 1
+        # Select action randomly or according to policy
+        if rpm.size() < args.start_timesteps:
+            action = env.action_space.sample()
+        else:
+            predict_action = agent.predict(np.array(obs))
+            action_noise = np.random.normal(
+                0, max_action * args.expl_noise, size=action_dim)
+            action = (predict_action + action_noise).clip(
+                -max_action, max_action)
+
+        # Perform action
+        next_obs, reward, done, _ = env.step(action)
+        terminal = float(done) if episode_steps < env._max_episode_steps else 0
+        terminal = 1. - terminal
+
+        # Store data in replay memory
+        rpm.append(obs, action, reward, next_obs, terminal)
+
+        obs = next_obs
+        episode_reward += reward
+
+        # Train agent after collecting sufficient data
+        if rpm.size() >= args.start_timesteps:
+            batch_obs, batch_action, batch_reward, batch_next_obs, batch_terminal = rpm.sample_batch(
+                args.batch_size)
+            agent.learn(batch_obs, batch_action, batch_reward, batch_next_obs,
+                        batch_terminal)
+
+    return episode_reward, episode_steps
+
+
+# Runs policy for 5 episodes by default and returns average reward
+# A fixed seed is used for the eval environment
+def run_evaluate_episodes(agent, env_name, seed, eval_episodes=5):
+    eval_env = gym.make(env_name)
+    eval_env.seed(seed + 100)
+
+    avg_reward = 0.
+    for _ in range(eval_episodes):
+        state, done = eval_env.reset(), False
+        while not done:
+            action = agent.predict(np.array(state))
+            state, reward, done, _ = eval_env.step(action)
+            avg_reward += reward
+    avg_reward /= eval_episodes
+    return avg_reward
+
+
+def main(args):
```

Comment:
args is a global var

Reply:
Resolved.
![image](https://user-images.githubusercontent.com/76139596/102741636-e36db680-438d-11eb-93e7-8574ee01c040.png)